### DISCUSSION TITLE
copy-webpack-plugin's error

### DISCUSSION THREAD

**Discussion by hellokey-key** [webpack/webpack#16284] [Ideas]
<img width="550" alt="image" src="https://user-images.githubusercontent.com/60806549/191720655-92a50e74-0ced-413b-9581-410e6d7838b3.png">
 you can see the Configuration，if you 'npm run build', the terminal will get error ' this.patterns.map()  Cannot read properties of undefined (reading 'map')' let us see the source of copy-webpack-plugin, it is show ```    this.patterns = options.patterns; ``` at line 185 , we console the result shows ``[ { patterns: [ [Object], [Object] ] } ]``, so it may be ``    this.patterns = options[0].patterns;`` is  perfect，Please see if it is correct. We look forward to your reply。

**Comment by hellokey-key** [comment]
<img width="500" alt="image" src="https://user-images.githubusercontent.com/60806549/191722122-7f846068-0ded-49cb-a719-f450f869be56.png">
<img width="708" alt="image" src="https://user-images.githubusercontent.com/60806549/191722173-cc795582-e874-4a4f-b7e6-8f61ccbbe431.png">

**Reply by alexander-akait** [reply]
Please check the version of the plugin

**Reply by hellokey-key** [reply]
the  plugin's version is ^11.0.0, but is the version for webpack is ^5.74.0 and webpack-cli is ^4.10.0, all is latest , why it  runs error?

**Reply by alexander-akait** [reply]
Can you provide small reproducible repo? thank you

**Reply by hellokey-key** [reply]
`const path = require('path')
const HtmlWebpackPlugin = require('html-webpack-plugin')
const { CleanWebpackPlugin } = require('clean-webpack-plugin')
const webpack = require('webpack');
const CopyWebpackPlugin = require('copy-webpack-plugin');

const RemoveWebpackPlugin = require('./src/remove-webpack-plugin');


module.exports = {
    mode: 'development',
    entry: {
        index: "./src/index.js",
        add: "./src/add.js",
        // css: './src/common.js'
    },
    output: {
        path: path.resolve(__dirname,'build'),
        filename: '[name].[chunkhash].js',
        publicPath: './'
    },
    devtool: "source-map",
    plugins:[
        new RemoveWebpackPlugin(),
        new CleanWebpackPlugin(),
        // new HtmlWebpackPlugin({
        //     filename: 'anout.html',
        //     template: './aout.html'
        // }),
        new HtmlWebpackPlugin({
            titie: 'title',
            template: './index.html',
            meta:{
                viewport: 'width=device-width, initial-scale=2.0'
            },
            filename: 'main.html',
            publicPath: './',
            scriptLoading: 'module'
        }),
        new CopyWebpackPlugin([
            {
                patterns:[
                    {
                        from: path.resolve(__dirname, './static'),
                        to: path.resolve(__dirname, 'build/static')
                    },
                ]
            }
        ])
    ],
    module:{
        rules:[
            {
                test:/\.css$/,
                use:[
                    'style-loader',
                    'css-loader'
                ]
            },
            {
                test:/\.md$/,
                use:[
                    {
                        loader: 'html-loader'
                    },
                    {
                        loader: './src/mkdown-loader.js',
                        options: {
                                author: 'hello loader',
                                email: 'helloloader@qq.com',
                                name: '张三' 
                        }
                    }
                ]
            }
        ]
    },
    devServer: {
        static: {
          directory: path.join(__dirname, 'build'),
        },
        compress: true,
        port: 9000,
        proxy: {
            "/api": {
                target: "https://apix.com/xx/xx",
                pathReWrite: {
                    "^/api":''
                },
                changeOrigin: true,
                hot: true
            },
        },
      },
   
}
`

<img width="316" alt="image" src="https://user-images.githubusercontent.com/60806549/192262526-9e967444-6256-4a05-8ee8-cbea6600e398.png">
by the way，the file directory is like this, which I added later and has nothing to do with the process，you can try it。

**Reply by alexander-akait** [reply]
Can you use github for this, because it is hard to read

**Reply by hellokey-key** [reply]
oh，i am sorry ,but you can copy these code into your vscode, let me see, you may write the same file to test。
